tap_zendesk/client.py

```diff
@@ -60,29 +67,20 @@ def http_headers(self) -> dict:
         headers = {}
         if "user_agent" in self.config:
             headers["User-Agent"] = self.config.get("user_agent")
-        # If not using an authenticator, you may also provide inline auth headers:
-        # headers["Private-Token"] = self.config.get("auth_token")  # noqa: ERA001
         return headers
 
-    def get_new_paginator(self) -> BaseAPIPaginator:
+    def get_new_paginator(self) -> JSONPathPaginator:
         """Create a new pagination helper instance.
 
-        If the source API can make use of the `next_page_token_jsonpath`
-        attribute, or it contains a `X-Next-Page` header in the response
-        then you can remove this method.
-
-        If you need custom pagination that uses page numbers, "next" links, or
-        other approaches, please read the guide: https://sdk.meltano.com/en/v0.25.0/guides/pagination-classes.html.
-
         Returns:
             A pagination helper instance.
         """
-        return super().get_new_paginator()
+        return JSONPathPaginator(self.next_page_token_jsonpath)
```

Comment:
Nice! 🚀

Reply:
This method can also be removed if we are using the `next_page_token_jsonpath`. See https://github.com/meltano/sdk/blob/db38fcdd9f88dfca67870f07122e9333c2ccfac7/singer_sdk/streams/rest.py#L538-L539